stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -122,26 +122,28 @@ public class ChannelListViewModel(
      * It connects the 'loadingMore', 'channelsState' and 'endOfChannels' properties from the [controller].
      */
     private suspend fun observeChannels(controller: QueryChannelsController) {
-        controller.channelsState.map { state ->
-            when (state) {
-                QueryChannelsController.ChannelsState.NoQueryActive,
-                QueryChannelsController.ChannelsState.Loading,
-                ->
-                    channelsState.copy(isLoading = true)
-                QueryChannelsController.ChannelsState.OfflineNoResults -> channelsState.copy(
-                    isLoading = false,
-                    channels = emptyList()
-                )
-                is QueryChannelsController.ChannelsState.Result -> {
-                    channelsState.copy(
+        controller.mutedChannelIds.combine(controller.channelsState) { mutedChannelIds, state -> mutedChannelIds to state }
```

Comment:
Keep in mind I'm going to deprecate `mutedChannelIds` this week (probably tomorrow). It's better to use `user.channelMutes` instead, but I'm not sure if we are updating it properly. I'll let you know once I'll finish the task